Source/WorkersES6/decodeGoogleEarthEnterprisePacket.js

```diff
@@ -201,15 +201,27 @@ function processMetadata(buffer, totalSize, quadKey) {
 function processTerrain(buffer, totalSize, transferableObjects) {
   var dv = new DataView(buffer);
 
+  // Each tile is split into 4 parts.
+  var advanceTile = function (pos) {
+    for (var quad = 0; quad < 4; ++quad) {
+      // Guard against reading past the end of buffer.
+      if (pos > totalSize) {
```

Comment:
This prevents us from reading a part of a tile that doesn't exist, but if the 4th part is incomplete, the returned value will be out of range and it is used in the `buffer.slice` call below.

Reply:
This block returns a -1, which is tested against below and will break out of the while loop.

Reply:
I cannot say if partial quads are valid. From the existing code, it appears they are not. I think you are concerned with a malformed buffer which is a valid point and I will adjust the code to test immediately after the getUint32() call instead of before it. However in practice, that is not the problem we are experiencing. There seems to be extra bytes in the buffer after all valid tiles have been extracted.

Reply:
After some experimentation with partial tiles, and examining code that uses uses these terrain tiles, I'm quite certain that partial tiles are not supported.
 
The method interpolateHeight() assumes the terrain buffer has 4 parts without ever checking for buffer overruns.

https://github.com/CesiumGS/cesium/blob/b3334483f6c543ce1f16a2e41d7c8c641f137567/Source/Core/GoogleEarthEnterpriseTerrainData.js#L466